### PR TITLE
Conditional rescan

### PR DESCRIPTION
**Pull request by fluidvanadium**
Conditional rescan if output indexes are missing!

### CODE REVIEW

**Comment by fluidvanadium** [comment]
this is not going to work so long as scan_full_tx is broken

File: zingolib/src/blaze/full_transactions_processor.rs [hunk 1]
```diff
@@ -21,6 +21,7 @@ use zcash_primitives::{
 
 use zingo_status::confirmation_status::ConfirmationStatus;
 
+// despite being called fetch_full_transaction, this function sends a txid somewhere else to be fetched as a Transaction. then processes it.
```
Comment (zancas) [inline zingolib/src/blaze/full_transactions_processor.rs:24]: This doc-comment seems to be misplaced?

Comment (zancas) [reply zingolib/src/blaze/full_transactions_processor.rs:24]: Is it referring to the second parameter?

Comment (fluidvanadium) [reply zingolib/src/blaze/full_transactions_processor.rs:24]: its referring the the mod, and its not a doc-comment

**Comment by fluidvanadium** [comment]
sorry, this solution has become unwieldy. adapting this PR to a new solution now

File: zingolib/src/wallet/transaction_context.rs [hunk 2]
```diff
@@ -31,6 +33,21 @@ impl TransactionContext {
...
+    /// returns any outdated records that need to be rescanned for completeness..
+    /// checks that each record contains output indexes for its notes
+    pub async fn unindexed_records(&self) -> Result<(), Vec<(TxId, BlockHeight)>> {
+        self.transaction_metadata_set
+            .read()
+            .await
+            .transaction_records_by_id
+            .get_spendable_note_ids_and_values(
+                &[ShieldedProtocol::Sapling, ShieldedProtocol::Orchard],
+                0.into(),
```
Comment (Oscar-Pepper) [inline zingolib/src/wallet/transaction_context.rs]: is this tested? transactions confirmed at or before 0 seems like it won't scan any transactions for missing output indices to me

Comment (Oscar-Pepper) [reply zingolib/src/wallet/transaction_context.rs]: should we change this to the height of the first block in the wallet? i.e. wallet height?

**Comment by Oscar-Pepper** [comment]
> this is not going to work so long as scan_full_tx is broken

im working on it!

**Review by Oscar-Pepper — APPROVED** [review]

**Comment by zancas** [comment]
This introduced 1 serious bug, that we almost didn't catch... and some unknown number of bugs that we are not aware of.

**Pull request merged**